build_tools/fetch_artifacts.py

```diff
@@ -0,0 +1,157 @@
+        "host-suite-sparse_lib",
+    ]
+    if args.blas or args.all:
+        base_artifacts.append("host-blas_lib")
+
+    for base_artifact in base_artifacts:
+        s3_exec(GENERIC_VARIANT, base_artifact, run_id, build_dir)
+
+
+def retrieve_enabled_artifacts(args, test_enabled, target, run_id, build_dir):
+    base_artifact_path = []
+    if args.blas or args.all:
+        base_artifact_path.append("blas")
+    if args.fft or args.all:
+        base_artifact_path.append("fft")
+    if args.miopen or args.all:
+        base_artifact_path.append("miopen")
+    if args.prim or args.all:
+        base_artifact_path.append("prim")
+    if args.rand or args.all:
+        base_artifact_path.append("rand")
+    if args.rccl or args.all:
+        base_artifact_path.append("rccl")
+
+    enabled_artifacts = []
+    for base_path in base_artifact_path:
+        enabled_artifacts.append(f"{base_path}_lib")
+        if test_enabled:
+            enabled_artifacts.append(f"{base_path}_test")
+
+    for enabled_artifact in enabled_artifacts:
+        s3_exec(f"{target}", enabled_artifact, run_id, build_dir)
+
+
+def run(args):
+    run_id = args.run_id
+    target = args.target
+    build_dir = args.build_dir
+    test_enabled = args.test
+    retrieve_base_artifacts(args, run_id, build_dir)
+    retrieve_enabled_artifacts(args, test_enabled, target, run_id, build_dir)
+
+
+def main(argv):
+    parser = argparse.ArgumentParser(prog="fetch_artifacts")
+    parser.add_argument(
+        "--run-id", type=str, help="GitHub run ID to retrieve artifacts from"
+    )
+
+    parser.add_argument(
+        "--target", type=str, help="Target variant for specific GPU target"
+    )
+
+    parser.add_argument(
+        "--build-dir",
+        type=str,
+        default="build/artifacts",
+        help="Path to the artifact build directory",
+    )
+
+    parser.add_argument(
+        "--test",
+        help="If flagged, test artifacts will be retrieved",
+        action="store_true",
+    )
+
+    parser.add_argument(
+        "--blas",
+        help="If flagged, blas artifacts will be retrieved",
+        action="store_true",
+    )
+
+    parser.add_argument(
+        "--fft", help="If flagged, fft artifacts will be retrieved", action="store_true"
+    )
+
+    parser.add_argument(
+        "--miopen",
+        help="If flagged, miopen artifacts will be retrieved",
+        action="store_true",
+    )
+
+    parser.add_argument(
+        "--prim",
+        help="If flagged, prim artifacts will be retrieved",
+        action="store_true",
+    )
+
+    parser.add_argument(
+        "--rand",
+        help="If flagged, rand artifacts will be retrieved",
+        action="store_true",
+    )
+
+    parser.add_argument(
+        "--rccl",
+        help="If flagged, rccl artifacts will be retrieved",
+        action="store_true",
+    )
+
+    parser.add_argument(
+        "--all", help="If flagged, all artifacts will be retrieved", action="store_true"
```

Comment:
I think you can get rid of this if you use BooleanOptionalArgument above and default them to True.